app/models/remote_authenticatable.rb

```diff
@@ -9,7 +9,8 @@ module RemoteAuthenticatable
       def remote_authentication(params)
         resp = authentication_flow(params)
 
-        if resp.response_type == AuthenticationBackend::CHALLENGE
+        if resp.response_type == AuthenticationBackend::CHALLENGE ||
```

Comment:
```suggestion
        if %w(AuthenticationBackend::CHALLENGE AuthenticationBackend::RETRY).include?(resp.response_type)
```
just a non-blocking suggestion